[Wang Peixuan] iP

wpx12011 opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/wpx12011/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

wpx12011 committed:
Level-1 Completed

wpx12011 committed:
Completed Level 2

wpx12011 committed:
Completed Text class

wpx12011 committed:
A-Iheritance

wpx12011 committed:
Completed Level-4

Commit:
Add TextUiTesting

Commit:
Completed Level-5

Commit:
Completed Level-6

wpx12011 committed:
Completed Level-7

Commit:
Completed Level-8

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Completed Level-8

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Completed Level-7

# Conflicts:
#	src/main/java/Duke.java

harin0826 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+                    } else if (fields[0].equals("E")) {
+                        currTask = new Event(fields[2], fields[3], fields[4]);
+                    } else {
+                        currTask = new Task("");
```
What would happen if text file was corrupted and fields[0] traversed through here? Should it still create the empty task?

harin0826 reviewed (commented):
Could utilize more OOP! Good work and good luck for iP! :)

axmszr commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.time.LocalTime;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
```
I think these import statements should be in alphabetical order! Like `.io` then `.time` then `util` (and the packages within each of those too). Checkstyle will be helpful in flagging it out :)

axmszr commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+        while (true) {
+            userInput = keyboard.next();
+            System.out.println("--------------------------------");
+            switch (userInput) {
```
I think the Java standard we follow doesn't have indentations for each `case`. No need for Egyptian braces within the cases too! Something like:
```java
switch (userInput) {
case "bye":
    ...
case "list":
    ...
...
```

axmszr commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+public class Deadline extends Task {
```
Should there be a blank line between the `import` statements and the Class header?

axmszr commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+            System.out.println("--------------------------------");
+        }
+    }
+}
```
Don't forget to add a blank line at the end of each file! I didn't remember to do it and checkstyle flagged all my files :P

axmszr commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException  extends Exception {
```
Should there be just one space?
```suggestion
public class DukeException extends Exception {
```

axmszr commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
```
Nice! This is really clean.

axmszr commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    public String toRecord() {
+        return (isDone ? "1|" : "0|") + description;
```
Ooh, this would like really nice in the save file! Should `Duke#takeRequest()` somehow handle the `"|"` character if it's given in a `userInput`?

axmszr commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,18 @@
+public class Todo extends Task {
+
+    protected String by;
```
Will `Todo` use this `by` variable in the future?

axmszr reviewed (commented):
Looks good so far! All the best with the next few iP tasks ^-^

Commit:
Completed A-MoreOOP

Commit:
Completed A-Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Completed A-Gradle

Commit:
Completed J-Unit

Commit:
Completed A-JavaDoc

Commit:
Completed A-JavaDoc

Commit:
Complete Level-9

Commit:
Merge branch 'branch-Level-9'

Commit:
Complete Level-10

Commit:
Complete A-Assertions

Add assertions in Parser class to ensure the parsed command name is not
empty.

Add assertions in Ui class to ensure the task index is within the length
of task list.

Commit:
Complete A-CodeQuality

Remove some unused methods in Ui class.

Assign variable to output of isValidCommand method in Parser class, to
make the code more obvious.

Modify output for "bye" command.

wpx12011 committed:
Merge pull request #2 from wpx12011/branch-A-Assertions

Complete A-Assertions

Add assertions in Parser class to ensure the parsed command name is not
empty.

Add assertions in Ui class to ensure the task index is within the length
of task list.

Commit:
Resolve merge conflicts

wpx12011 committed:
Merge branch 'master' into branch-A-CodeQuality

wpx12011 committed:
Merge pull request #3 from wpx12011/branch-A-CodeQuality

Complete A-CodeQuality

Commit:
Complete Level BCD-Extension

Implement [help] command which display sample commands to users

Commit:
Complete A-UserGuide

Add user guide and ui.png

Commit:
Modify Ui.PNG

Commit:
Modify Ui.png
